modules/3d/include/opencv2/3d.hpp

```diff
@@ -2342,6 +2341,167 @@ class CV_EXPORTS CvLevMarq
     int solveMethod;
 };
 
+
+/** @brief Octree for 3D vision.
+In 3D vision filed, the Octree is used to process and accelerate the pointcloud data. The class Octree represents
+the Octree data structure. Each Octree will have a fixed depth. The depth of Octree refers to the distance from
+the root node to the leaf node.All OctreeNodes will not exceed this depth.Increasing the depth will increase
+the amount of calculation exponentially. And the small number of depth refers low resolution of Octree.
+Each node contains 8 children, which are used to divide the space cube into eight parts. Each octree node represents
+a cube. And these eight children will have a fixed order, the order is described as follows:
+
+    For illustration, assume,
+        rootNode: origin == (0, 0, 0), size == 2
+     Then,
+        children[0]: origin == (0, 0, 0), size == 1
+        children[1]: origin == (1, 0, 0), size == 1, along X-axis next to child 0
+        children[2]: origin == (0, 1, 0), size == 1, along Y-axis next to child 0
+        children[3]: origin == (1, 1, 0), size == 1, in X-Y plane
+        children[4]: origin == (0, 0, 1), size == 1, along Z-axis next to child 0
+        children[5]: origin == (1, 0, 1), size == 1, in X-Z plane
+        children[6]: origin == (0, 1, 1), size == 1, in Y-Z plane
+        children[7]: origin == (1, 1, 1), size == 1, furthest from child 0
+*/
+class Octree {
+
+public:
+
+    //! Default constructor.
+    Octree();
+
+    /** @overload
+        * @brief Create an empty Octree and set the maximum depth.
+        * @param _maxDepth The max depth of the Octree. The _maxDepth > -1.
+        */
+    Octree(int _maxDepth);
```

Comment:
Avoid using underscore prefixes in public headers.
We don't need them in the public API.

BTW, C++ allows to use any name for parameters in .cpp file (but it should correlate with name in .hpp to keep code readable).

Reply:
Thanks, this will be fixed in the new version.